### DISCUSSION TITLE
Illegal character in module-path

### DISCUSSION THREAD

**Discussion by freixas** [wiverson/jtoolprovider-plugin#12] [Q&A] [answered]
I'm just starting to try to leverage all your hard work. I get this when I try to run mvn clean install:

[INFO] --- jtoolprovider-plugin:1.0.34:java-tool (jlink) @ gamma ---
[ERROR] jlink failed with error code [1]
[ERROR]    --add-modules
[ERROR]    org.freixas.gamma
[ERROR]    --module-path
[ERROR]    D:\Users\tony\Documents\Java\gamma/org.freixas.gamma,D:\Users\tony\Documents\Java\gamma/javafx-windows-x86_64/javafx-jmods-17.0.2/"
[ERROR]    --output
[ERROR]    D:\Users\tony\Documents\Java\gamma\target/jvm-image
[ERROR]    --strip-native-commands
[ERROR]    --no-header-files
[ERROR]    --strip-debug
[ERROR]    --no-man-pages
[ERROR]    --compress=2
[INFO] Error: Illegal char <:> at index 54: D:\Users\tony\Documents\Java\gamma/org.freixas.gamma,D:\Users\tony\Documents\Java\gamma/javafx-windows-x86_64/javafx-jmods-17.0.2/"

In the POM, I have:

<modulePath>
    ${project.basedir}/org.freixas.gamma,${project.basedir}/javafx-${os.detected.classifier}/javafx-jmods-${javafx.version}/"
</modulePath>

It's odd that ":" is an illegal character. How do I specify the drive? 

Also, I guess I'm ignorant, but I thought jmod info would be part of the the javafx dependency in my .m2 folder. I could have sworn I saw them there, but they aren't there now (or anywhere). But they are definitely not in my project basedir. Oops, it looks like I need to set these up--that's not part of the maven-jpackage-template instructions, but I see the files are in your template.

I use commons-cli as an unamed module, so it's going to be fun to see how that gets included.

Right now, my software goes out as a ZIP file and users have to manually install Java and JavaFX, Once that's done and a couple of environment variables are defined, they can execute my code with:

java -p %JAVAFX_HOME%;%GAMMA_HOME% --add-modules org.freixas.gamma -jar %GAMMA_HOME%\gamma.jar

org.freixas.gamma pulls everything else in. But I gather the <modulePath> is not the same as -p.

Sorry, I know Java from back in the Java V1 days, but I've only recently had to deal with Git, Maven, JavaFX, Java modules, etc. My software works great, but getting an installable package for the major platforms is sure a giant pain!

**Comment by wiverson** [comment, marked as answer]
On mobile right now, quick thought, use the windows path character not a comma to separate the path entry

**Comment by freixas** [comment]
Thanks. I'll check this out. I think I had the problem before I had the "org.freixas.gamma," part, but I'm not 100% sure. Like you, I am not at my computer right now. I'll try to check this out tonight, but am leaving on a short trip, so it may take a few days. I'll try to add the jmod files as well. You might as well wait for me to check these things out so as not to waste your time.

**Comment by freixas** [comment]
You were right, silly me. The module-path is the same as the -p argument to java and so it was staring me in the face (it's hard to see between the two %'s).
